openpype/hosts/flame/plugins/publish/collect_timeline_instances.py

```diff
@@ -34,119 +34,124 @@ class CollectTimelineInstances(pyblish.api.ContextPlugin):
-                subset = marker_data["subset"]
-
-                # insert family into families
-                family = marker_data["family"]
-                families = [str(f) for f in marker_data["families"]]
-                families.insert(0, str(family))
-
-                # form label
-                label = asset
-                if asset != clip_name:
-                    label += " ({})".format(clip_name)
-                label += " {}".format(subset)
-                label += " {}".format("[" + ", ".join(families) + "]")
-
-                inst_data.update({
-                    "name": "{}_{}".format(asset, subset),
-                    "label": label,
-                    "asset": asset,
-                    "item": segment,
-                    "families": families,
-                    "publish": marker_data["publish"],
-                    "fps": self.fps,
-                    "flameSourceClip": source_clip,
-                    "sourceFirstFrame": int(first_frame),
-                    "path": file_path
-                })
-
-                # get otio clip data
-                otio_data = self._get_otio_clip_instance_data(clip_data) or {}
-                self.log.debug("__ otio_data: {}".format(pformat(otio_data)))
-
-                # add to instance data
-                inst_data.update(otio_data)
-                self.log.debug("__ inst_data: {}".format(pformat(inst_data)))
-
-                # add resolution
-                self._get_resolution_to_data(inst_data, context)
-
-                # add comment attributes if any
-                inst_data.update(comment_attributes)
-
-                # create instance
-                instance = context.create_instance(**inst_data)
-
-                # add colorspace data
-                instance.data.update({
-                    "versionData": {
-                        "colorspace": clip_data["colour_space"],
-                    }
-                })
-
-                # create shot instance for shot attributes create/update
-                self._create_shot_instance(context, clip_name, **inst_data)
-
-                self.log.info("Creating instance: {}".format(instance))
-                self.log.info(
-                    "_ instance.data: {}".format(pformat(instance.data)))
-
-                if not with_audio:
-                    continue
+        for segment in selected_segments:
+            comment_attributes = self._get_comment_attributes(segment)
+            self.log.debug("__ segment.name: {}".format(
+                segment.name
+            ))
+            self.log.debug("_ comment_attributes: {}".format(
+                pformat(comment_attributes)))
+
+            clip_data = opfapi.get_segment_attributes(segment)
+            clip_name = clip_data["segment_name"]
+            self.log.debug("clip_name: {}".format(clip_name))
+
+            # get openpype tag data
+            marker_data = opfapi.get_segment_data_marker(segment)
```

Comment:
I would say that these conditions should be first step of the loop before getting comments and clip data.
```
for segment in selected_segments:
    marker_data = opfapi.get_segment_data_marker(segment)
    self.log.debug("__ marker_data: {}".format(
        pformat(marker_data)))

    if not marker_data:
        continue

    if marker_data.get("id") != "pyblish.avalon.instance":
        continue

    comment_attributes = self._get_comment_attributes(segment)
    ...
```

Reply:
Ah yeah, you are right.